apps/web/src/components/activity/ExecutionDetailTrigger.tsx

```diff
@@ -11,16 +11,34 @@ const TriggerTitle = styled(Text)`
   padding-bottom: 20px;
 `;
 
-const buildTrigger = (identifier, subscriberId, payload): INotificationTrigger => ({
-  type: TriggerTypeEnum.EVENT,
-  identifier,
-  variables: payload ? Object.entries(payload).map(([name, value]) => ({ name, value })) : [],
-  subscriberVariables: [{ name: 'subscriberId', value: subscriberId }],
-});
-
-export const ExecutionDetailTrigger = ({ step }) => {
-  const { identifier, payload, _subscriberId } = step || {};
-  const trigger = buildTrigger(identifier, _subscriberId, payload);
+const buildTrigger = (identifier, subscriber, payload): INotificationTrigger => {
+  const subscriberVariables = [{ name: 'subscriberId', value: subscriber._id }];
+
+  if (subscriber.email) {
```

Comment:
So we should add here for firstName, lastName, and maybe even avatar. As they are all variables that can be used in the template for a subscriber
What do you think?

Reply:
Oh wait. But we only want to show in the trigger the subscriberVariables that were actually sent - so we keep the data for debugging accurate. Also, if subscriber doesn't have a phone but a phone is needed and sent in the to payload - that's a situation that will also want to be shown. Or if he sent in the trigger a different phone.

Reply:
@ainouzgali, does the query brings from DB subscriber variables with information that might be not used in the initial call?

Reply:
Yes, you understood me what I mean. But then that means we have no way to exactly know all the subscriber variables used on the trigger. I can't think how we could identify them.

Reply:
We do save the subscriberVariables inside the trigger in the template - so we kind of have a way to get those. 
But about the values, I'm not completely sure where they would be if not from the subscriber.